code/modules/power/tesla/energy_ball.dm

```diff
@@ -0,0 +1,203 @@
+#define TESLA_DEFAULT_POWER 3476520
+#define TESLA_MINI_POWER 1738260
+
+var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
+										/obj/machinery/power/emitter,
+										/obj/machinery/field/generator,
+										/mob/living/simple_animal,
+										/obj/machinery/particle_accelerator/control_box,
+										/obj/structure/particle_accelerator/fuel_chamber,
+										/obj/structure/particle_accelerator/particle_emitter/center,
+										/obj/structure/particle_accelerator/particle_emitter/left,
+										/obj/structure/particle_accelerator/particle_emitter/right,
+										/obj/structure/particle_accelerator/power_box,
+										/obj/structure/particle_accelerator/end_cap,
+										/obj/machinery/containment_field,
+										/obj/structure/disposalpipe)
+
+/obj/singularity/energy_ball
+	name = "energy eall"
+	desc = "An energy ball."
+	icon = 'icons/obj/tesla_engine/energy_ball.dmi'
+	icon_state = "energy_ball"
+	pixel_x = -32
+	pixel_y = -32
+	current_size = STAGE_TWO
+	move_self = 1
+	grav_pull = 0
+	contained = 0
+	density = 1
+	var/list/orbiting_balls = list()
+	var/produced_power
+	var/is_orbiting
+
+/obj/singularity/energy_ball/Destroy()
+	for(var/obj/singularity/energy_ball/EB in orbiting_balls)
+		qdel(EB)
+	return ..()
+
+/obj/singularity/energy_ball/process()
+	if(!is_orbiting)
+		handle_energy()
+		var/amount_to_move = 2 + orbiting_balls.len * 2
+		move_the_basket_ball(amount_to_move)
+		pixel_x = 0
+		pixel_y = 0
+		tesla_zap(src, 7, TESLA_DEFAULT_POWER)
+		pixel_x = -32
+		pixel_y = -32
+		energy += rand(1,3) // ensure it generates energy without needing to be blasted by the PA too much due to its size, and that a tesla engine will always get bigger over time
+	else
+		tesla_zap(src, 7, TESLA_MINI_POWER)
+		energy = 0 // ensure we dont have miniballs of miniballs
+	return
+
+/obj/singularity/energy_ball/examine(mob/user)
+	..()
+	if(orbiting_balls.len)
+		user << "The amount of orbiting mini-balls is [orbiting_balls.len]."
+
+
+/obj/singularity/energy_ball/proc/move_the_basket_ball(var/move_amount)
+	for(var/i = 0, i < move_amount, i++)
```

Comment:
I hope you don't mind me commenting on this after closure, but I really want to see this IG. Anyway, where is move_amount even set, because as it is right now, it is a useless check in the middle parameter (i < move_amount), because it is only defined in the proc parameter and never used nor set.
BTW, if I'm talking jibberish, I am sorry, my DM is real rusty.